sdk/src/network/mod.rs

```diff
@@ -116,6 +121,196 @@ pub trait Network: Send + Sync + Clone + 'static {
     }
 }
 
+pub struct NetworkMultiplexer {
+    to_receiving_streams: ActiveStreams,
+    unclaimed_receiving_streams: Arc<DashMap<StreamKey, MultiplexedReceiver>>,
+    tx_to_networking_layer: MultiplexedSender,
+}
+
+type ActiveStreams = Arc<DashMap<StreamKey, tokio::sync::mpsc::UnboundedSender<ProtocolMessage>>>;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Default, Serialize, Deserialize)]
+pub struct StreamKey {
+    pub job_id: u8,
```

Comment:
This job id has nothing to do with jobs correct? Agnostic P2P network to anything blueprint related.

Reply:
It doesn't have to. In fact, we can remove it and solely depend on the task_id for uniqueness.

Reply:
Ok let's remove it.